### PR TITLE
Fix discount prices not respecting `showZeroDecimalPlacePrices`

### PR DESCRIPTION
**Pull request by vegaro**
### Motivation

Follow-up to #6242. The `productOfferPrice` and `productOfferPricePerDay/Week/Month/Year` functions in `VariableHandlerV2` returned discount prices without applying the `showZeroDecimalPlacePrices` formatting. For example, a $2.00 intro offer would always show as "$2.00" even when `showZeroDecimalPlacePrices` was enabled (which should display "$2").

Regular product price functions correctly used `isPriceEndingIn00Cents`/`formatAsZeroDecimalPlaces` but the discount code paths skipped this logic.

### Description

- Add a `formattedPrice` helper in the private `VariablesV2` extension that checks if a price ends in .00 and strips the decimal places when `showZeroDecimalPlacePrices` is true
- Apply this helper in all 5 offer price functions: `productOfferPrice`, `productOfferPricePerDay`, `productOfferPricePerWeek`, `productOfferPricePerMonth`, `productOfferPricePerYear`
- Add test data `packageWithIntroOfferPayUpFrontWholeDollar` (a $60/year product with a $2.00 pay-up-front intro offer)
- Add 4 tests verifying discount prices respect the zero-decimal-places flag

### CODE REVIEW

File: RevenueCatUI/Helpers/PriceFormatterExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        showZeroDecimalPlacePrices: Bool
+    ) -> String {
+        guard showZeroDecimalPlacePrices else {
+            return priceString
+        }
+
+        guard let price = self.number(from: priceString)?.doubleValue else {
+            return priceString
+        }
+
```
Comment (facumenzella) [inline RevenueCatUI/Helpers/PriceFormatterExtensions.swift:34]: Nit: Floating point equality can be unreliable. Consider using `Decimal` comparison instead to avoid edge cases like `2.00` becoming `1.9999999999998`.

File: RevenueCatUI/Helpers/PriceFormatterExtensions.swift [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        _ priceString: String,
+        showZeroDecimalPlacePrices: Bool
+    ) -> String {
+        guard showZeroDecimalPlacePrices else {
+            return priceString
+        }
+
+        guard let price = self.number(from: priceString)?.doubleValue else {
+            return priceString
+        }
```
Comment (facumenzella) [inline RevenueCatUI/Helpers/PriceFormatterExtensions.swift:30]: This assumes 2 decimal places, but not all currencies work that way (e.g., JPY has 0, KWD has 3). Could we derive this from the formatter's `maximumFractionDigits` instead?

File: RevenueCatUI/Templates/V2/Variables/VariableHandlerV2.swift [hunk 3]
```diff
@@ -770,7 +777,12 @@ extension VariablesV2 {
             return ""
         }
 
-        return formatter.string(from: price as NSDecimalNumber) ?? ""
+        guard let priceString = formatter.string(from: price as NSDecimalNumber) else {
+            return ""
+        }
+
+        return formatter.priceStringWithZeroDecimalFormatting(
```
Comment (facumenzella) [inline RevenueCatUI/Templates/V2/Variables/VariableHandlerV2.swift]: This pattern is repeated in all 5 offer price functions. Would be cleaner to extract a helper like `formatOfferPrice(price:formatter:showZeroDecimal:)`.

File: RevenueCatUI/Helpers/PriceFormatterExtensions.swift [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+extension NumberFormatter {
+
+    func priceStringWithZeroDecimalFormatting(
+        _ priceString: String,
+        showZeroDecimalPlacePrices: Bool
+    ) -> String {
+        guard showZeroDecimalPlacePrices else {
+            return priceString
+        }
+
```
Comment (facumenzella) [inline RevenueCatUI/Helpers/PriceFormatterExtensions.swift:27]: Since the caller already has the raw `Decimal` price, we could avoid this string→number→string round-trip by accepting the price directly.

File: RevenueCatUI/Helpers/PriceFormatterExtensions.swift [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PriceFormatterExtensions.swift
+
+import Foundation
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, *)
+extension NumberFormatter {
+
```
Comment (facumenzella) [inline RevenueCatUI/Helpers/PriceFormatterExtensions.swift:16]: Minor: The name suggests it *will* format with zero decimals, but it conditionally does so. Maybe `formattedPriceStrippingTrailingZerosIfNeeded` or similar?

File: RevenueCatUI/Helpers/PriceFormatterExtensions.swift (hunk 1 above)
Comment (vegaro) [reply RevenueCatUI/Helpers/PriceFormatterExtensions.swift:34]: Good call! Replaced `Double` comparison with `Decimal` arithmetic using `NSDecimalRound` — the `isWholeNumber` check now uses exact decimal comparison, avoiding any floating-point edge cases.

File: RevenueCatUI/Templates/V2/Variables/VariableHandlerV2.swift (hunk 3 above)
Comment (vegaro) [reply RevenueCatUI/Templates/V2/Variables/VariableHandlerV2.swift]: Done! Extracted `formatDiscountPrice(_:package:showZeroDecimalPlacePrices:fallback:)` in the private extension. All 5 offer price functions now delegate to it, with `productOfferPrice` passing `discount.localizedPriceString` as the fallback and the per-period functions using the default empty string fallback.

File: RevenueCatUI/Helpers/PriceFormatterExtensions.swift (hunk 4 above)
Comment (vegaro) [reply RevenueCatUI/Helpers/PriceFormatterExtensions.swift:27]: Fixed! The primary overload now accepts `Decimal` directly — the VariableHandlerV2 callers pass `discount.price` or `discount.pricePerDay?.decimalValue` etc. without any string round-trip. A convenience `from priceString:` overload is kept for `Package+VariableDataProvider` callers that only have the formatted string.

File: RevenueCatUI/Helpers/PriceFormatterExtensions.swift (hunk 5 above)
Comment (vegaro) [reply RevenueCatUI/Helpers/PriceFormatterExtensions.swift:16]: Renamed to `formattedPriceStrippingTrailingZerosIfNeeded` — should be clearer that it conditionally strips zeros rather than always formatting with zero decimals.

File: RevenueCatUI/Helpers/PriceFormatterExtensions.swift (hunk 2 above)
Comment (vegaro) [reply RevenueCatUI/Helpers/PriceFormatterExtensions.swift:30]: Actually, to clarify: `showZeroDecimalPlacePrices` is specifically about showing prices with zero decimal places when the price is a whole number (e.g. "$2.00" → "$2"). So we are not trying to respect the currency's native decimal count here — we just need to check whether the price is a whole number. The old `round(price * 100) / 100.0` was a crude "is this whole?" check, not a currency-aware rounding. The `Decimal.isWholeNumber` fix addresses the real concern (floating-point reliability) without needing `maximumFractionDigits`.
